### PR TITLE
refactor: upgrade deno_ast 0.28 and deno_semver 0.4

### PR DESCRIPTION
**Pull request by dsherret**

### CODE REVIEW

File: cli/resolver.rs [hunk 1]
```diff
@@ -8,14 +8,14 @@ use deno_core::futures::future::LocalBoxFuture;
 use deno_core::futures::FutureExt;
 use deno_core::ModuleSpecifier;
 use deno_core::TaskQueue;
-use deno_graph::source::NpmPackageReqResolution;
 use deno_graph::source::NpmResolver;
+use deno_graph::source::PackageReqResolution;
```
Comment (dsherret) [inline cli/resolver.rs:12]: This rename was actually a mistake because this won't be used for deno: resolution. I will revert it later.

File: cli/tests/testdata/npm/cjs_with_deps/main_info.out [hunk 2]
```diff
@@ -4,19 +4,19 @@ dependencies: 14 unique
...
-  └── npm:type-detect@4.0.8 ([WILDCARD])
+├─┬ npm:/chalk@4.1.2 ([WILDCARD])
+│ ├─┬ npm:/ansi-styles@4.3.0 ([WILDCARD])
+│ │ └─┬ npm:/color-convert@2.0.1 ([WILDCARD])
+│ │   └── npm:/color-name@1.1.4 ([WILDCARD])
+│ └─┬ npm:/supports-color@7.2.0 ([WILDCARD])
+│   └── npm:/has-flag@4.0.0 ([WILDCARD])
+└─┬ npm:/chai@4.3.6 ([WILDCARD])
+  ├── npm:/assertion-error@1.1.0 ([WILDCARD])
+  ├── npm:/check-error@1.0.2 ([WILDCARD])
```
Comment (bartlomieju) [inline cli/tests/testdata/npm/cjs_with_deps/main_info.out:15]: Why this change?

File: cli/tests/testdata/npm/invalid_package_name/main.out [hunk 3]
```diff
@@ -1,2 +1,2 @@
-error: Not a valid package: @foo
+error: Invalid package specifier 'npm:@foo'. Did not contain a valid package name.
```
Comment (bartlomieju) [inline cli/tests/testdata/npm/invalid_package_name/main.out:1]: Very nice 👍

**Review by bartlomieju — APPROVED** [review]

File: cli/tests/testdata/npm/cjs_with_deps/main_info.out (hunk 2 above)
Comment (dsherret) [reply cli/tests/testdata/npm/cjs_with_deps/main_info.out:15]: The resolved specifier for an npm module now includes the `/`, so the top level specifiers displaying here were also having the foward slash:

```
file:///V:/deno/cli/tests/testdata/npm/cjs_with_deps/main.js (325B)
├─┬ npm:/chalk@4.1.2 (34.23KB)
│ ├─┬ npm:ansi-styles@4.3.0 (16.58KB)
│ │ └─┬ npm:color-convert@2.0.1 (26.55KB)
│ │   └── npm:color-name@1.1.4 (6.54KB)
│ └─┬ npm:supports-color@7.2.0 (6.87KB)
│   └── npm:has-flag@4.0.0 (4.32KB)
└─┬ npm:/chai@4.3.6 (732.97KB)
  ├── npm:assertion-error@1.1.0 (5.51KB)
  ...
```

To make it consistent I added the forward slash to everything. I think it's more correct?

Comment (bartlomieju) [reply cli/tests/testdata/npm/cjs_with_deps/main_info.out:15]: Yeah, I think it makes sense, but does that mean that we will require `npm:/chai` in the import statement as well? I guess not

Comment (dsherret) [reply cli/tests/testdata/npm/cjs_with_deps/main_info.out:15]: It won't be required

**Comment by tilemanrenovations** [comment]
Use tinker for a toolbox to make the code easier to understand and run denoted by this user is a good friend so it won't panic as much.
